### ISSUE TITLE
Improve visual accessibility of text links

### ISSUE THREAD

**Issue by mmbotelho** [okTurtles/group-income#514]
### Problem

All text links in the app today rely solely on color.

### Solution

We should add an underline, as it is a common pattern accross the web - meaning that users are more than familiar with it at this point. For more information (and arguments for and against) on this topic, read [Guidelines for Visualizing Links](https://www.nngroup.com/articles/guidelines-for-visualizing-links/).

I went ahead and designed a possible solution for this. I used a `border-bottom` instead of `text-decoration: underline;`, because the latter often renders in a weird way (example in the comments). All colors should be used as they are now (normal state, on hover, on focus, etc). The border uses $primary-light instead of $link. Here's how it looks:

<img width="830" alt="captura de ecra 2019-01-08 as 16 37 11" src="https://user-images.githubusercontent.com/22004566/50840860-bae87f80-1363-11e9-8332-80b201110752.png">

**Comment by mmbotelho** [comment]
On the difference between `text-decoration: underline` and `border-bottom`:
<img width="574" alt="captura de ecra 2019-01-08 as 16 47 35" src="https://user-images.githubusercontent.com/22004566/50841511-2d0d9400-1365-11e9-9647-a2502d4dabee.png">

The first one has gaps whenever the font has a descender (in this case the `p`, `(` and `)` characters). This creates a lot of visual noise.

**Comment by mmbotelho** [comment]
@pieer you can find the Figma component here https://www.figma.com/file/ZR0DRPhhUuHdJtf7PbGAM6tW/Group-Income?node-id=48%3A3
